### DISCUSSION TITLE
removing a mesh of points from another mesh

### DISCUSSION THREAD

**Discussion by cyubit** [pyvista/pyvista#4243] [Q&A]
I have a mesh of only points (red) that I want to remove from another mesh (green), How should I go about doing this? Boolean difference requires both meshes to be triangulated, and remove_points returns "TypeError: Remove must be either a mask or an integer array-like" when I try mesh.remove_points(points)

Here they are overlayed 
<img width="403" alt="Screen Shot 2023-04-06 at 11 22 42 PM" src="https://user-images.githubusercontent.com/87676972/230534577-2c327893-7542-4535-9727-ede001bc62cf.png">

**Comment by banesullivan** [comment]
I'm confused on what result you're looking for... 

You could try to find the closest points in the mesh to each of those red points: https://docs.pyvista.org/api/core/_autosummary/pyvista.DataSet.find_closest_point.html#find-closest-point

Then use those indices to remove those points from the mesh: https://docs.pyvista.org/api/core/_autosummary/pyvista.PolyDataFilters.remove_points.html#remove-points

But this seems very odd to me and I'm struggling to make sense of the result that would produce

--- 

Or are you trying to cut holes in the mesh? If this, you may need to generate a geometry (sphere) for each point of the desired hole size, then boolean operation those meshes

**Reply by cyubit** [reply]
I'm trying to remove the inside surface of this curvy and irregularly shaped mesh. My current approach involves ray tracing from the center of the mesh to all the points in the mesh and taking the first collision of each ray. This creates a set of all the points on the inside surface of the mesh (plus a few more), which I am then trying to remove from the mesh to ultimately cut it in half kind of like a clam shell. 

Initial mesh
<img width="345" alt="Screen Shot 2023-04-07 at 10 29 10 AM" src="https://user-images.githubusercontent.com/87676972/230625825-04c29180-88c3-4d82-be5c-bbe4f75b2ee5.png">

The ray tracing step 
<img width="462" alt="Screen Shot 2023-04-07 at 10 22 00 AM" src="https://user-images.githubusercontent.com/87676972/230624820-824e1074-1ef0-411d-990d-31f5fb830130.png">

the goal (with no inside surface)
<img width="310" alt="Screen Shot 2023-04-07 at 10 28 18 AM" src="https://user-images.githubusercontent.com/87676972/230625698-89cb4a94-5947-4abc-8c04-7e108c6198d0.png">

**Reply by cyubit** [reply]
Actually, I used your approach and figured it out! Turns out I was too sleepy to notice that remove_points actually removes point indices and not coordinates. I've been trying to remove the list of coordinates this whole time but once I noticed that find_closest_point returned point indices it clicked. Thank you very much!

**Reply by banesullivan** [reply]
Ah, I see. Perhaps thresholding on those intersection cells would work better? There have been some similar examples in the discussions before you may be able to dig up.

The ray tracing should return the cell IDs of all intersections, you could use that to make a mask to then `threshold()` the mesh by.


hope this helps
